File: src/kyu_7/partial_word_searching/js/wordSearch.js
function wordSearch(query, seq){
    const strings = [];
    for (const s of seq) {
        if (s.toLowerCase().includes(query.toLowerCase())) {
            strings.push(s);
        }
    }
    return strings.length !== 0 ? strings : ["Empty"];
}
module.exports = wordSearch;

File: src/kyu_7/partial_word_searching/java/WordSearch.java
package kyu_7.partial_word_searching.java;

import java.util.ArrayList;
import java.util.List;

class WordSearch {
    static String[] findWord(String x, String[] y) {
        List<String> strings = new ArrayList<>();
        for (String s : y) {
            if (s.toLowerCase().contains(x)) {
                strings.add(s);
            }
        }
        return strings.toArray(new String[0]).length != 0 ? strings.toArray(new String[0]) : new String[]{"Empty"};
    }
}